File: 0841-keys-and-rooms/0841-keys-and-rooms.java
class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        int visited[] = new int[rooms.size()];
        // visited[0] = 1;
        int count = 0;
        for(int i = 0 ; i < rooms.size() ; i++){
            if(visited[i] == 0){
                dfs(i, rooms, visited);
                count++;
            }
        }
        if(count > 1) return false;
        return true;

    }
    void dfs(int node, List<List<Integer>> adj, int[] visited){
        visited[node] = 1;
        for(int i = 0 ; i < adj.get(node).size() ; i++){
            if(visited[adj.get(node).get(i)] != 1)
            dfs(adj.get(node).get(i), adj, visited);
        }
    }
}

File: 0841-keys-and-rooms/0841-keys-and-rooms.cpp
class Solution {
public:
    bool canVisitAllRooms(vector<vector<int>>& rooms) {
        queue<int> que;
        vector<int> visited(rooms.size(), 0);
        que.push(0);
        while(!que.empty()){
            int curr = que.front();
            visited[curr] = 1;
            que.pop();
            for(int i = 0 ; i < rooms[curr].size() ; i++){
                if(visited[rooms[curr][i]] == 0)
                que.push(rooms[curr][i]);
            }

        }
        for(int i = 0 ; i < rooms.size() ; i++){
            if(visited[i] == 0){
                return false;
            }
        }
        return true;
    }
};